### PR TITLE
Deprecate dwavebinarycsp

### PR DESCRIPTION
**Pull request by arcondello**
I didn't add messages to all of the docstrings, but I did add messages pretty much everywhere else.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,9 +1,10 @@
+:warning: *dwavebinarycsp* is deprecated. For solving problems with constraints,
+    we recommend using the hybrid solvers in the Leap service. You can find
```
Comment (JoelPasvolsky) [inline README.rst:2]:
```suggestion
    we recommend using the hybrid solvers in the Leap&trade; service. You can find
```

First use for the code

File: README.rst [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
+    documentation for the hybrid solvers at https://docs.ocean.dwavesys.com.
+
 .. image:: https://img.shields.io/pypi/v/dwavebinarycsp.svg
     :target: https://pypi.org/project/dwavebinarycsp
 
-.. image:: https://ci.appveyor.com/api/projects/status/b99rhw0l6ljsgw5t?svg=true
```
Comment (arcondello) [inline README.rst:4]: This was just no longer relevant, technically unrelated but :shrug:

File: docs/index.rst [hunk 3]
```diff
@@ -6,6 +6,13 @@
 D-Wave Binary CSP
 =================
 
+.. deprecated:: 0.3.1
+
+    ``dwavebinarycsp`` is deprecated and will be removed in Ocean 10.
+    For solving problems with constraints, we recommand using the hybrid
+    solvers in the Leap service.
```
Comment (randomir) [inline docs/index.rst:13]: Can we recommend (or link to) a non-HSS solution? For someone wanting to say in the QPU-land. Or that wouldn't make sense?

Comment (arcondello) [reply docs/index.rst:13]: IMO for constrained problems we want to steer them to the HSS. There are some alternatives when using the QPU directly but those are rather specific, and I don't think those users would be coming from here.

Comment (JoelPasvolsky) [inline docs/index.rst:13]:
```suggestion
    solvers in the Leap&trade; service.
```

Import note: change this over to [sdk_index file](https://docs.ocean.dwavesys.com/en/stable/docs_binarycsp/sdk_index.html) for the SDK

File: docs/index.rst [hunk 4]
```diff
@@ -6,6 +6,13 @@
...
+    You can find documentation for the hybrid solvers at :ref:using_hybrid.
```
Comment (JoelPasvolsky) [inline docs/index.rst:14]:
```suggestion
    You can find documentation for the hybrid solvers at :ref:`using_hybrid`.
```

File: docs/intro.rst [hunk 5]
```diff
@@ -4,6 +4,13 @@
 Introduction
 ============
 
+.. deprecated:: 0.3.1
+
+    ``dwavebinarycsp`` is deprecated and will be removed in Ocean 10.
+    For solving problems with constraints, we recommand using the hybrid
+    solvers in the Leap service.
+    You can find documentation for the hybrid solvers at :ref:using_hybrid.
```
Comment (JoelPasvolsky) [inline docs/intro.rst:12]:
```suggestion
    You can find documentation for the hybrid solvers at :ref:`using_hybrid`.
```

File: dwavebinarycsp/compilers/stitcher.py [hunk 6]
```diff
@@ -12,6 +12,8 @@
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
 
+import warnings
```
Comment (randomir) [inline dwavebinarycsp/compilers/stitcher.py:15]: Why is this imported if never used?

File: docs/reference/compilers.rst [hunk 7]
```diff
@@ -4,6 +4,13 @@
 Converting to a Binary Quadratic Model
 ======================================
 
+.. deprecated:: 0.3.1
+
+    ``dwavebinarycsp`` is deprecated and will be removed in Ocean 10.
+    For solving problems with constraints, we recommand using the hybrid
+    solvers in the Leap service.
+    You can find documentation for the hybrid solvers at :ref:using_hybrid.
```
Comment (JoelPasvolsky) [inline docs/reference/compilers.rst:12]:
```suggestion
    You can find documentation for the hybrid solvers at :ref:`using_hybrid`.
```

File: docs/reference/constraint.rst [hunk 8]
```diff
@@ -4,6 +4,13 @@
 Defining Constraints
 ====================
 
+.. deprecated:: 0.3.1
+
+    ``dwavebinarycsp`` is deprecated and will be removed in Ocean 10.
+    For solving problems with constraints, we recommand using the hybrid
+    solvers in the Leap service.
+    You can find documentation for the hybrid solvers at :ref:using_hybrid.
```
Comment (JoelPasvolsky) [inline docs/reference/constraint.rst:12]:
```suggestion
    You can find documentation for the hybrid solvers at :ref:`using_hybrid`.
```

File: dwavebinarycsp/compilers/stitcher.py (hunk 6 above)
Comment (arcondello) [reply dwavebinarycsp/compilers/stitcher.py:15]: oops, it was briefly. Will remove.

File: docs/reference/csp.rst [hunk 9]
```diff
@@ -4,6 +4,13 @@
 Defining Constraint Satisfaction Problems
 =========================================
 
+.. deprecated:: 0.3.1
+
+    ``dwavebinarycsp`` is deprecated and will be removed in Ocean 10.
+    For solving problems with constraints, we recommand using the hybrid
+    solvers in the Leap service.
+    You can find documentation for the hybrid solvers at :ref:using_hybrid.
```
Comment (JoelPasvolsky) [inline docs/reference/csp.rst:12]:
```suggestion
    You can find documentation for the hybrid solvers at :ref:`using_hybrid`.
```

File: docs/reference/factory_constraints.rst [hunk 10]
```diff
@@ -4,6 +4,13 @@
 Factories
 =========
 
+.. deprecated:: 0.3.1
+
+    ``dwavebinarycsp`` is deprecated and will be removed in Ocean 10.
+    For solving problems with constraints, we recommand using the hybrid
+    solvers in the Leap service.
+    You can find documentation for the hybrid solvers at :ref:using_hybrid.
```
Comment (JoelPasvolsky) [inline docs/reference/factory_constraints.rst:12]:
```suggestion
    You can find documentation for the hybrid solvers at :ref:`using_hybrid`.
```

File: docs/reference/loading.rst [hunk 11]
```diff
@@ -4,6 +4,13 @@
 Other CSP Formats
 =================
 
+.. deprecated:: 0.3.1
+
+    ``dwavebinarycsp`` is deprecated and will be removed in Ocean 10.
+    For solving problems with constraints, we recommand using the hybrid
+    solvers in the Leap service.
+    You can find documentation for the hybrid solvers at :ref:using_hybrid.
```
Comment (JoelPasvolsky) [inline docs/reference/loading.rst:12]:
```suggestion
    You can find documentation for the hybrid solvers at :ref:`using_hybrid`.
```

File: docs/reference/reduction.rst [hunk 12]
```diff
@@ -4,6 +4,13 @@
 Reducing Constraints
 ====================
 
+.. deprecated:: 0.3.1
+
+    ``dwavebinarycsp`` is deprecated and will be removed in Ocean 10.
+    For solving problems with constraints, we recommand using the hybrid
+    solvers in the Leap service.
+    You can find documentation for the hybrid solvers at :ref:using_hybrid.
```
Comment (JoelPasvolsky) [inline docs/reference/reduction.rst:12]:
```suggestion
    You can find documentation for the hybrid solvers at :ref:`using_hybrid`.
```

File: README.rst (hunk 1 above)
Comment (arcondello) [reply README.rst:2]: Will that work in GH flavoured markdown?

**Comment by JoelPasvolsky** [comment]
> Shouldn't the deprecation not also be added to `sdk_index.rst`?

Yes, see [this comment](https://github.com/dwavesystems/dwavebinarycsp/pull/124#discussion_r1861309491)

File: README.rst (hunk 1 above)
Comment (arcondello) [reply README.rst:2]: https://stackoverflow.com/questions/12753022/how-do-you-escape-the-trademark-or-copyright-symbol-in-markdown

File: docs/index.rst (hunk 3 above)
Comment (arcondello) [reply docs/index.rst:13]: Added it to the README but not the docs after discussion with @JoelPasvolsky

**Review by thisac — APPROVED** [review]
LGTM

**Review by JoelPasvolsky — APPROVED** [review]
Thanks!

**Review by randomir — APPROVED** [review]

**Pull request merged**